### ISSUE TITLE
Look at routing data to determine if standardization is possible, without limiting proprietary route engins/solvers.

### ISSUE THREAD

**Issue by ayoumans** [opengeospatial/ideas#64]

**Comment by jeffharrison** [comment]
This would likely be useful to many communities. Perhaps a simple first step might be developing a JSON encoding for exchanging a Route (?). After that perhaps the basic elements of a simple REST API...

**Comment by prushforth** [comment]
Would love to see someone express a route as MapML features, displayed on a MapML web map.

**Comment by jeffharrison** [comment]
That is a very interesting idea. So would a 'Route MapML feature' go directly into an HTML document in the <layer-> element (?)

**Comment by jeffharrison** [comment]
... in the '<layer> element' ;-)

**Comment by jeffharrison** [comment]
... 'in the layer element'. 

Note to self, don't use angle brackets on this forum ;-)

**Comment by prushforth** [comment]
> Note to self, don't use angle brackets on this forum ;-)

If you want to use angle brackets or other markup, I learned the hard way that you need to use the little angle bracket icon to create backtics in which you paste your code/angle brackets.  If you want a multiline code snippet like a whole xml document, you use triple backtics for some reason I don't understand, but if you select the text and then use the angle bracket icon tool above, it will automagically put the appropriate backtics on the selection.

Anyhew, I was thinking that a search widget could be incorporated into the controls of a map as a search input in a "static" map layer.   The idea of static map layers was recently floated (by me) in the Testbed 14 MapML [repo](https://github.com/opengeospatial/D012-MapML_Engineering_Report/issues/14).  But to summarize, it would be a layer that would contain controls and other HTML markup that should be treated as a non-panning non-zooming layer, generally on the top of the stack of layers, of course.  There is sort of implicitly such a layer in a `<map>` now, although it is not reflected in the DOM as such.  But, to get back to the idea, the search text input could be directed by the map document author to request results in whatever format, preferably MapML.  As such the results could be displayed automatically on the map if / when selected by the user.  I guess a route is a kind of search result.  So, perhaps it could be represented as MapML features.

Anyway, perhaps its worth exploring in a testbed or some such activity.

**Comment by jeffharrison** [comment]
Yes, whether a Route could be represented as a MapML feature is definitely worth exploring in a Testbed.
